contracts/Gauge.sol

```diff
@@ -2,7 +2,9 @@
 pragma solidity 0.8.13;
 
 import "./interfaces/IExtraReward.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin-upgradeable/contracts/token/ERC20/ERC20Upgradeable.sol";
 import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import "@openzeppelin/contracts/utils/math/Math.sol";
```

Comment:
I would also use the contracts in the `penzeppelin-contracts-upgradeable` dependency